pramen/core/src/main/scala/za/co/absa/pramen/core/utils/impl/ResultSetToRowIterator.scala

```diff
@@ -112,13 +112,27 @@ class ResultSetToRowIterator(rs: ResultSet, sanitizeDateTime: Boolean, incorrect
 
     // WARNING. Do not forget that `null` is a valid value returned by RecordSet methods that return a reference type objects.
     dataType match {
-      case BIT | BOOLEAN => rs.getBoolean(columnIndex)
-      case TINYINT       => rs.getByte(columnIndex)
-      case SMALLINT      => rs.getShort(columnIndex)
-      case INTEGER       => rs.getInt(columnIndex)
-      case BIGINT        => rs.getLong(columnIndex)
-      case FLOAT         => rs.getFloat(columnIndex)
-      case DOUBLE        => rs.getDouble(columnIndex)
+      case BIT | BOOLEAN =>
+        val v = rs.getBoolean(columnIndex)
+        if (rs.wasNull()) null else v
```

Comment:
Very ugly, but it seems it is the way to go, at least for the maximum performance

Reply:
```scala
val rsValue = dataType match {
  case BIT | BOOLEAN => rs.getBoolean(columnIndex)
  case TINYINT =>  rs.getByte(columnIndex)
  case SMALLINT      => rs.getShort(columnIndex)
  case INTEGER       => rs.getInt(columnIndex)
  case BIGINT        => rs.getLong(columnIndex)
  case FLOAT         => rs.getFloat(columnIndex)
  case DOUBLE        => rs.getDouble(columnIndex)        
  case REAL          => rs.getBigDecimal(columnIndex)
  case NUMERIC       => rs.getBigDecimal(columnIndex)
  case DATE          => sanitizeDate(rs.getDate(columnIndex))
  case TIMESTAMP     => sanitizeTimestamp(rs.getTimestamp(columnIndex))
  case _             => rs.getString(columnIndex)
}
if (rs.wasNull()) null else rsValue
```

Reply:
Yes, this looks much nicer indeed! Just it would call `rs.wasNull()` even when it is not necessary, so I still prefer the original approach.